packages/attestation-service/src/sms/twilio.ts

```diff
@@ -76,15 +119,76 @@ export class TwilioSmsProvider extends SmsProvider {
     } catch (error) {
       throw new Error(`Twilio Messaging Service could not be fetched: ${error}`)
     }
+    if (this.verifyServiceSid) {
+      try {
+        await this.client.verify.services
+          .get(this.verifyServiceSid)
+          .fetch()
+          .then((service) => {
+            if (!service.customCodeEnabled) {
+              // Make sure that custom code is enabled
+              throw new Error(
+                'TWILIO_VERIFY_SERVICE_SID is specified, but customCode is not enabled. Please contact Twilio support to enable it.'
+              )
+            }
+          })
+      } catch (error) {
+        throw new Error(`Twilio Verify Service could not be fetched: ${error}`)
+      }
+    }
   }
 
   async sendSms(attestation: AttestationModel) {
-    const m = await this.client.messages.create({
-      body: attestation.message,
-      to: attestation.phoneNumber,
-      from: this.messagingServiceSid,
-      statusCallback: this.deliveryStatusURL,
-    })
-    return m.sid
+    // Prefer Verify API if Verify Service is present and
+    // user is not requesting a deeplink
+    if (this.verifyServiceSid && attestation.securityCode) {
+      const requestParams: any = {
+        to: attestation.message,
+        channel: 'sms',
+        customCode: attestation.securityCode,
+      }
+
+      if (attestation.appSignature) {
+        requestParams.appHash = attestation.appSignature
+      }
+      // Normalize to locales that Twilio supports
+      // If locale is not supported, Twilio API will throw an error
+      if (attestation.language) {
+        const locale = attestation.language.toLocaleLowerCase()
+        if (['es-419', 'es-us', 'es-la'].includes(locale)) {
+          attestation.language = 'es'
+        }
+        if (this.twilioSupportedLocales.includes(locale)) {
+          requestParams.locale = locale
+        }
+      }
+      try {
+        const m = await this.client.verify
+          .services(this.verifyServiceSid)
+          .verifications.create(requestParams)
+        return m.sid
+      } catch (e) {
+        // Verify landlines using voice
+        if (e.message.includes('SMS is not supported by landline phone number')) {
+          requestParams.appHash = undefined
+          requestParams.channel = 'call'
```

Comment:
this feels like a poor API design...they should auto-recognized and change to 'call' if its landline 🙄

perhaps once spruce work is merged, we can select the appropriate method on first try.

Reply:
Yeah, I thought there'd be a config in the portal to do this automatically but it's not the case. On the bright side, the Verify API returns the phone number type so we can avoid making the type API call altogether. We'd still need exception clause, but a vast majority of users (mobile) would avoid the double API call.